packages/terra-form-select/src/Select.jsx

```diff
@@ -0,0 +1,297 @@
+    return (event) => {
+      this.handleSelection(event, option);
+
+      if (option.props.onClick) {
+        option.props.onClick(event);
+      }
+    };
+  }
+
+  wrapOnKeyDown(option) {
+    return (event) => {
+      if (option.props.onKeyDown) {
+        option.props.onKeyDown(event);
+      }
+
+      if (event.nativeEvent.keyCode === SelectUtils.KEYCODES.SPACE || event.nativeEvent.keyCode === SelectUtils.KEYCODES.ENTER) {
+        this.handleSelection(event, option);
+      } else if (event.nativeEvent.keyCode === SelectUtils.KEYCODES.UP_ARROW) {
+        let previousFocus = event.target.previousSibling;
+        while (previousFocus && previousFocus.tabIndex < 0) {
+          previousFocus = previousFocus.previousSibling;
+          if (!previousFocus) {
+            return;
+          }
+        }
+        previousFocus.focus();
+      } else if (event.nativeEvent.keyCode === SelectUtils.KEYCODES.DOWN_ARROW) {
+        let nextFocus = event.target.nextSibling;
+        while (nextFocus && nextFocus.tabIndex < 0) {
+          nextFocus = nextFocus.nextSibling;
+          if (!nextFocus) {
+            return;
+          }
+        }
+        nextFocus.focus();
+      }
+    };
+  }
+
+  handleSelection(event, option) {
+    if (this.props.onChange && option.props.value !== this.state.value) {
+      this.props.onChange(event, option.props.value);
+    }
+
+    let newValue = null;
+    if (!this.props.value) {
+      newValue = option.props.value;
+    }
+
+    this.setState({ selectedValue: newValue, isOpen: false });
+  }
+
+  handleOnClick() {
+    this.setState({ isOpen: true });
+  }
+
+  handleOnKeyDown(event) {
+    if (event.nativeEvent.keyCode === SelectUtils.KEYCODES.SPACE
+      || event.nativeEvent.keyCode === SelectUtils.KEYCODES.UP_ARROW
+      || event.nativeEvent.keyCode === SelectUtils.KEYCODES.DOWN_ARROW) {
+      this.setState({ isOpen: true });
+    }
+  }
+
+  handleOnRequestClose() {
+    this.setState({ isOpen: false });
+  }
+
+  render() {
+    const {
+      onChange,
+      name,
+      required,
+      defaultValue,
+      value,
+      isInvalid,
+      disabled,
+      children,
+      requestFocus,
+      releaseFocus,
+      ...customProps
+    } = this.props;
+
+    const { intl } = this.context;
+
+    const attributes = Object.assign({}, customProps);
+    const selectClasses = cx([
+      'select',
+      { invalid: isInvalid },
+      { 'is-disabled': disabled },
+      { open: this.state.isOpen },
+      attributes.className,
+    ]);
+
+    if (required) {
+      attributes['aria-required'] = 'true';
+    }
+
+    attributes.tabIndex = disabled ? '-1' : '0';
+    attributes['aria-disabled'] = disabled;
+
+    let selectedValue;
+    let display;
+    let defaultOption;
+    let isDefaultDisplay = false;
+    const clonedChildren = React.Children.map(children, (child) => {
+      let isSelected = false;
+
+      if (child.props.value === value || child.props.value === this.state.selectedValue) {
```

Comment:
Could we add an inline comment saying something like 'check if the option is the selected option'. It took me a second to realize why this wasn't a separate function to wrap the children and that this helped determine if the defaultDisplay should be created.

Reply:
Added comments: https://github.com/cerner/terra-core/pull/978/commits/70812139c85b71e4035c0388406c73685de32e1f